robocup_ws/src/Planner/src/player_controller.py

```diff
@@ -0,0 +1,128 @@
+from game_interfaces.msg import PlayerCommand
+import numpy as np
+import matplotlib.pyplot as plt
+from BasicCommonActions.go_to_action import simple_go_to_action
+from game_interfaces.msg import Position
+from Planner.src.brachistochrone import cycloid
+from BasicCommonActions.go_to_action import go_to_fast
+import math
+
+
+class PlayerController:  #(Robot)
+    def __init__(self, player_id):
+        self.read = True
+        self.player = player_id
+        self.points_to_visit = []
+        self.current_goal = 0
+        self.goal_threshold = 0.1
+
+    def has_ball(self, game_info):
+        team = game_info[0]
+        player_id = game_info[2]
+        position = team.players_positions_efcs[player_id]
+        ball_pos = team.ball_pos_efcs
+        d = np.hypot(ball_pos.x - position.x, ball_pos.y - position.y)
+        if (d <= self.goal_threshold):
+            return True
+        return False
+
+    def if_able_score(self, game_info):
+        pass
+
+    def check_for_pass(self, game_info):
```

Comment:
Add docstring

There are two functions which maybe do the same: if so, reuse the code or replace existing one if this one is better

https://github.com/szgula/UofG_Robotics_TDP/blob/a7fddf38f5438ca2089b1e7bdb62ef6a77113c7a/robocup_ws/src/Planner/src/BasicCommonActions/plan_supporting_functions.py#L146

 https://github.com/szgula/UofG_Robotics_TDP/blob/a7fddf38f5438ca2089b1e7bdb62ef6a77113c7a/robocup_ws/src/Planner/src/BasicCommonActions/plan_supporting_functions.py#L137